File: app/src/main/java/algorithm/study/dohyun/programmers/Prg_42862.java
package algorithm.study.dohyun.programmers;
import java.util.*;
public class Prg_42862 {
    public static void main(String[] args){

        Prg_42862 sol = new Prg_42862();
        int n = 5;
        int[] lost = {2,4};
        int[] reserve = {1,3,5};

        System.out.println(sol.solution(n,lost,reserve));
        System.out.println(sol.solution(5,new int[] {2,4}, new int[] {3}));
    }
    public int solution(int n, int[] lost, int[] reserve){

        ArrayList<Integer> lostlist = new ArrayList<>();
        Arrays.sort(lost);
        Arrays.sort(reserve);

        for(int i: lost)
            lostlist.add(i);
        for( int i = 0; i< reserve.length ; i++){
            if( lostlist.contains(reserve[i]) ){
                lostlist.remove(lostlist.indexOf(reserve[i]));
                reserve[i] = -1;
            }
        }

        for( int i = 0; i< reserve.length; i++){
            int man = reserve[i];

            if( lostlist.contains(man-1))
                lostlist.remove(lostlist.indexOf(man-1));
            else if( lostlist.contains( man+1 ) )
                lostlist.remove(lostlist.indexOf(man+1));

        }

        return n - lostlist.size() ;
    }
}

File: app/src/main/java/algorithm/study/devjk_sample/programmers/Prg_42862.kt
package algorithm.study.devjk_sample.programmers

/**
 * 프로그래머스 - 체육복
 *
 * @link https://school.programmers.co.kr/learn/courses/30/lessons/42862
 */
class Prg_42862 {

    fun solution(n: Int, lost: IntArray, reserve: IntArray): Int {
        val arr = Array(n + 1) { 1 }
        for (num in lost) {
            arr[num]--
        }
        for (num in reserve) {
            arr[num]++
        }

        for (i in arr.indices) {
            if (i == 0) continue

            val cur = arr[i]
            if (cur != 2) {
                continue
            }
            val left = i - 1
            if (left >= 1 && arr[left] == 0) {
                arr[left] = 1
                arr[i] = 1
                continue
            }

            val right = i + 1
            if (right < arr.size && arr[right] == 0) {
                arr[right] = 1
                arr[i] = 1
                continue
            }

        }

        var ans = 0
        for (i in arr.indices) {
            if (i == 0) {
                continue
            }
            if (arr[i] > 0) {
                ans++
            }
        }
        return ans
    }

}

fun main(args: Array<String>) {
    val sol = Prg_42862()
    println(sol.solution(5, intArrayOf(2, 4), intArrayOf(1, 3, 5)))
    println(sol.solution(5, intArrayOf(2, 4), intArrayOf(3)))
    println(sol.solution(3, intArrayOf(3), intArrayOf(1)))
}